src/coreclr/src/vm/reflectioninvocation.cpp

```diff
@@ -2174,6 +1997,269 @@ lExit: ;
+
+    // Don't allow ref structs
+    if (pMT->IsByRefLike())
+    {
+        COMPlusThrow(kNotSupportedException, W("NotSupported_ByRefLike"));
+    }
+}
+
+/*
+ * Given a RuntimeType, queries info on how to instantiate the object.
+ * pRuntimeType - [required] the RuntimeType object
+ * ppfnAllocator - [required, null-init] fnptr to the allocator
+ *                 mgd sig: void* -> object
+ * pvAllocatorFirstArg - [required, null-init] first argument to the allocator
+ *                       (normally, but not always, the MethodTable*)
+ * ppfnCtor - [required, null-init] the instance's parameterless ctor,
+ *            mgd sig object -> void, or null if no ctor is needed for this type
+ * pfCtorIsPublic - [required, null-init] whether the parameterless ctor is public
+ * ==========
+ * This method will not run the type's static cctor.
+ * This method will not allocate an instance of the target type.
+ */
+void QCALLTYPE RuntimeTypeHandle::GetActivationInfo(
+    QCall::ObjectHandleOnStack pRuntimeType,
+    PCODE* ppfnAllocator,
+    void** pvAllocatorFirstArg,
+    PCODE* ppfnCtor,
+    BOOL* pfCtorIsPublic
+)
+{
+    CONTRACTL{
+        QCALL_CHECK;
+        PRECONDITION(CheckPointer(ppfnAllocator));
+        PRECONDITION(CheckPointer(pvAllocatorFirstArg));
+        PRECONDITION(CheckPointer(ppfnCtor));
+        PRECONDITION(CheckPointer(pfCtorIsPublic));
+        PRECONDITION(*ppfnAllocator == NULL);
+        PRECONDITION(*pvAllocatorFirstArg == NULL);
+        PRECONDITION(*ppfnCtor == NULL);
+        PRECONDITION(*pfCtorIsPublic == NULL);
+    }
+    CONTRACTL_END;
+
+    TypeHandle typeHandle = NULL;
+
+    // If we're activating __ComObject.
+    void* pClassFactory = NULL;
+
+    BEGIN_QCALL;
+
+    {
+        GCX_COOP();
+
+        // We need to take the RuntimeType itself rather than the RuntimeTypeHandle,
+        // as the COM CLSID is stored in the RuntimeType object's sync block, and we
+        // might need to pull it out later in this method.
+        typeHandle = ((REFLECTCLASSBASEREF)pRuntimeType.Get())->GetType();
+    }
+
+    ValidateTypeAbleToBeInstantiated(typeHandle, false /* fGetUninitializedObject */);
+
+    MethodTable* pMT = typeHandle.AsMethodTable();
+    PREFIX_ASSUME(pMT != NULL);
+
+#ifdef FEATURE_COMINTEROP
+    // COM allocation can involve the __ComObject base type (with attached CLSID) or a
+    // VM-implemented [ComImport] class. For GetUninitializedObject, we block all COM types.
+    // For CreateInstance, the flowchart is:
+    //   - For __ComObject,
+    //     .. on Windows, bypass normal newobj logic and use ComClassFactory::CreateInstance.
+    //     .. on non-Windows, treat as a normal class, type has no special handling in VM.
+    //   - For [ComImport] class, treat as a normal class. VM will replace default
+    //     ctor with COM activation logic on supported platforms, else ctor itself will PNSE.
+    if (pMT->IsComObjectType())
+    {
+        if (IsComObjectClass(typeHandle))
```

Comment:
```suggestion
    if (IsComObjectClass(typeHandle))
    {
```
IsComObjectClass includes the check for IsComObjectType

Reply:
Ah, right, this is a holdover from when this block had code paths to handle both `__ComObject` vs. some arbitrary `[ComImport]`-annotated type. (It was for error checking that was since refactored away.)

Yeah, this can be simplified.